sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsConf.scala

```diff
@@ -936,6 +936,17 @@ object RapidsConf {
       .booleanConf
       .createWithDefault(false)
 
+  val OPTIMIZER_MODE = conf("spark.rapids.sql.optimizer.mode")
```

Comment:
I had considered making the optimizer a plugin, and that would let me move some of this outside the main repo, at least until we get to a point where we think it might have value for customers. I will work on this.